core/commands/block.go

```diff
@@ -185,3 +188,99 @@ func getBlockForKey(req cmds.Request, skey string) (blocks.Block, error) {
 	log.Debugf("ipfs block: got block with key: %q", b.Key())
 	return b, nil
 }
+
+var blockRmCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Remove IPFS block(s).",
+		ShortDescription: `
+'ipfs block put' is a plumbing command for removing raw ipfs blocks.
+It takes a list of base58 encoded multihashs to remove.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("hash", true, true, "Bash58 encoded multihash of block(s) to remove."),
+	},
+	Options: []cmds.Option{
+		cmds.BoolOption("ignore-pins", "Ignore pins.").Default(false),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		ignorePins, _, err := req.Option("ignore-pins").Bool()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		hashes := req.Arguments()
+		keys := make([]key.Key, 0, len(hashes))
+		for _, hash := range hashes {
+			k := key.B58KeyDecode(hash)
+			keys = append(keys, k)
+		}
+		rdr, wtr := io.Pipe()
+		go func() {
+			pinning := n.Pinning
+			if ignorePins {
+				pinning = nil
+			}
+			err := rmBlocks(n.Blockstore, pinning, wtr, keys)
+			if err != nil {
+				wtr.CloseWithError(fmt.Errorf("Some blocks not deleted: %s", err))
+			} else {
+				wtr.Close()
+			}
+		}()
+		res.SetOutput(rdr)
```

Comment:
Think about it this way, if you use `CloseWithError` that manifests as a failed call to `Read` somewhere. What sort of errors would you expect to prevent you from reading a stream of data? Examples where we use this are with `ipfs cat`, if we fail to fetch some of the data for ipfs cat, it will throw an error on the reader, which makes sense because you are failing to read the file youre catting.

In this case, youre failing to read a stream of data because you failed to delete a block? Those feel very unrelated to eachother.